File: Data Structures & Algorithms/car-fleet/submission-2.py
class Solution:
    def carFleet(self, target: int, position: List[int], speed: List[int]) -> int:
        d = {} # (pos, speed)
        fleet = 1
        for i in range(len(position)):
            d[position[i]] = speed[i]
        d = dict(sorted(d.items(), reverse=True))

        stack = []
        for pos, speed in d.items():
            timeToTarget = (target - pos) / speed
            if not stack:
                stack.append(timeToTarget)
            elif stack[-1] < timeToTarget:
                stack.append(timeToTarget)
        return len(stack)

File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.cs
public class Solution {
    public int[] TwoSum(int[] numbers, int target) {
        int l = 0;
		int r = numbers.Length - 1;
		
		while (l < r) {
			int currSum = numbers[l] + numbers[r];
			
			if (currSum < target) {
				l++;
			}
			else if (currSum > target) {
				r--;
			}
			else {
				break;
			}
		}
		
		return new int[] {l + 1, r + 1};
    }
}
